tests/e2e/lifecycle/mcg/test_mcg_namespace_lifecyle_crd.py

```diff
@@ -266,3 +269,181 @@ def test_mcg_namespace_lifecycle_crd(
+    )
+    @tier2
+    @skipif_ocs_version("<4.8")
+    @pytest.mark.polarion_id("OCS-2471")
+    def test_mcg_cache_lifecycle(
+        self, mcg_obj, cld_mgr, awscli_pod, bucket_factory, bucketclass_dict
+    ):
+        """
+        Test MCG cache bucket lifecycle
+
+        1. Create cache buckets on each namespace stores (RGW-OBC/OBC)
+        2. Verify write operations cache and hub bucket
+        3. Verify read/list operations on cache bucket and hub target
+        4. Verify delete operation on buckets
+        5. Delete multiple cache buckets with data still in ns store
+        6. Recreate the cache buckets on ns store(with existing data) then read.
+
+        """
+        data = "Sample string content to write to a S3 object"
+        object_key = "ObjKey-" + str(uuid.uuid4().hex)
+        if (
+            constants.RGW_PLATFORM
+            in bucketclass_dict["namespace_policy_dict"]["namespacestore_dict"]
+        ):
+            s3_creds = {
+                "access_key_id": cld_mgr.rgw_client.access_key,
+                "access_key": cld_mgr.rgw_client.secret_key,
+                "endpoint": cld_mgr.rgw_client.endpoint,
+            }
+            logger.info("RGW obc will be created as cache bucket")
+            obc_interface = "rgw-oc"
+        else:
+            s3_creds = {
+                "access_key_id": cld_mgr.aws_client.access_key,
+                "access_key": cld_mgr.aws_client.secret_key,
+                "endpoint": constants.MCG_NS_AWS_ENDPOINT,
+                "region": config.ENV_DATA["region"],
+            }
+            logger.info("Noobaa obc will be created as cache bucket")
+            obc_interface = bucketclass_dict["interface"]
+
+        # Create the namespace resource and bucket
+        ns_bucket = bucket_factory(
+            interface=obc_interface,
+            bucketclass=bucketclass_dict,
+        )[0]
+        logger.info(f"Cache bucket: {ns_bucket.name} created")
+        target_bucket = ns_bucket.bucketclass.namespacestores[0].uls_name
+
+        # Write to cache
+        logger.info(f"Writing object on cache bucket: {ns_bucket.name}")
+        assert s3_put_object(
+            mcg_obj, ns_bucket.name, object_key, data
+        ), "Failed: PutObject"
+        wait_for_cache(mcg_obj, ns_bucket.name, [object_key])
+
+        # Write to hub and read from cache
+        logger.info("Setting up test files for upload")
+        setup_base_objects(awscli_pod, amount=3)
+        logger.info(f"Uploading objects to ns target: {target_bucket}")
+        sync_object_directory(
+            awscli_pod,
+            src=MCG_NS_ORIGINAL_DIR,
+            target=f"s3://{target_bucket}",
+            signed_request_creds=s3_creds,
+        )
+        sync_object_directory(
+            awscli_pod, f"s3://{ns_bucket.name}", MCG_NS_RESULT_DIR, mcg_obj
+        )
+
+        # Read cached object
+        assert s3_get_object(mcg_obj, ns_bucket.name, object_key), "Failed: GetObject"
```

Comment:
Please create better assertion messages for get object failure instead of `Failed: GetObject` throughout the test.